integration/update_test.go

```diff
@@ -466,6 +467,98 @@ func TestUpdateSet(t *testing.T) {
 	}
 }
 
+func TestUpdateTimestamp(t *testing.T) {
+	t.Parallel()
+
+	// store the current timestamp with $currentDate operator;
+	// read it, check that it is close to the current time;
+	// write a new timestamp value with the same time;
+	// read it back, and check that it is still close to the current time.
+	t.Run("currentDateReadBack", func(t *testing.T) {
+		maxDifference := time.Duration(2 * time.Second)
+		now := time.Now()
+		nowTimestamp := primitive.Timestamp{T: uint32(now.Unix()), I: uint32(0)}
+		id := "string-empty"
+
+		tc := struct {
+			id     string
+			update bson.D
+			result bson.D
+			paths  []types.Path
+			stat   *mongo.UpdateResult
+			alt    string
+		}{
+			id:     id,
+			update: bson.D{{"$currentDate", bson.D{{"value", bson.D{{"$type", "timestamp"}}}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			paths:  []types.Path{types.NewPathFromString("value")},
+			result: bson.D{{"_id", id}, {"value", nowTimestamp}},
+		}
+
+		ctx, collection := setup(t, shareddata.Scalars, shareddata.Composites)
+
+		res, err := collection.UpdateOne(ctx, bson.D{{"_id", tc.id}}, tc.update)
+		require.NoError(t, err)
+		require.Equal(t, tc.stat, res)
+
+		var actualB bson.D
+		err = collection.FindOne(ctx, bson.D{{"_id", tc.id}}).Decode(&actualB)
+		require.NoError(t, err)
+
+		expected := ConvertDocument(t, tc.result)
+		actualX := ConvertDocument(t, actualB)
+
+		for _, path := range tc.paths {
+			testutil.CompareAndSetByPathTime(t, expected, actualX, maxDifference, path)
+		}
+
+		// set timestamp manually
+		updateB := bson.D{{"$set", bson.D{{"value", nowTimestamp}}}}
+		expectedB := bson.D{{"_id", id}, {"value", nowTimestamp}}
+		res, err = collection.UpdateOne(ctx, bson.D{{"_id", tc.id}}, updateB)
+		require.NoError(t, err)
+		require.Equal(t, tc.stat, res)
+
+		err = collection.FindOne(ctx, bson.D{{"_id", id}}).Decode(&actualB)
+		require.NoError(t, err)
+
+		AssertEqualDocuments(t, expectedB, actualB)
+		actualY := ConvertDocument(t, actualB)
+		testutil.CompareAndSetByPathTime(t, actualY, actualX, maxDifference, types.NewPathFromString("value"))
+	})
+
+	// store bson.Timestamp
+	// read it back, and check if values equal
+	t.Run("timestamp", func(t *testing.T) {
+		t.Parallel()
+
+		ctx, collection := setup(t)
+
+		now := time.Now()
+		nowTimestamp := primitive.Timestamp{T: uint32(now.Unix()), I: uint32(0)}
```

Comment:
I think this test does not test anything that is not covered by the previous test. Here we write some value and later get it back. It is not important for that test that this value is a Unix timestamp

Reply:
> I think this test does not test anything that is not covered by the previous test. Here we write some value and later get it back. It is not important for that test that this value is a Unix timestamp

removed.